src/qibolab/instruments/qblox/sequencer.py

```diff
@@ -127,6 +130,7 @@ def bake_pulse_waveforms(
         # there may be other waveforms stored already, set first index as the next available
         first_idx = len(self.unique_waveforms)
 
+        #
```

Comment:
Baking the pulses for flux or non-flux requires minimal differences, branching at top-level is not worth it (since it cause enough duplication).

Reply:
Moreover, the memory assessment is once more interleaved with the baking, and nested even deeper.

It is even handled differently, `NotEnoughMemoryForBaking`, that is unhandled. But it is not clear to me that this **has to** be managed differently, i.e. all the baked pulses for a duration sweeper should lend on the same sequencer, while the other are allowed to do differently (but I should better inspect how `pulse.idx_range` is used).